client/src/auth.js

```diff
@@ -127,7 +127,19 @@ export class TokenStorage {
 
   // Whether there is an auth token for the provided authType
   hasAuthToken() {
-    return Boolean(this.get())
+    const token = this.get();
+    if(!Boolean(token)){
+      return false;
+    }
+    try {
+      const meta = JSON.parse(atob(token.split('.')[1]));
```

Comment:
If you run eslint it will call out a few formatting errors like extraneous semicolons